### PR TITLE
Interface style improvement

### PR DESCRIPTION
**Pull request by kamilritz**
This PR contains several non-functional improvements regarding style:
- Replace some private float arrays with matrix::Vectors
- Change interface functions to output by return-by-value
- Add const modifier to getters
- Enforce camelCase in changed functions

The flash size is decreasing by roughly 250 Bytes for v4/v5 targets, but for v2 the flash is overflowing by a tiny bit.
During profiling a SITL mission flight, no significant performance difference has been detected.

This is changing the interface. It could be an option to keep the old interface and mark it as deprecated.

Firmware PR: https://github.com/PX4/Firmware/pull/14162

Testing:
- No functional change indicated
- FixedWind [Log](https://review.px4.io/plot_app?log=862caccc-2651-4703-bd9b-65e6406f6d52)
- Multicopter [Log](https://review.px4.io/plot_app?log=288e6bf6-9e3e-48b9-956d-d14a8cb700fd)
- Log fields touched by the interface changes show reasonable data.

TODO:
- [x] Profiling
- [x] Check log files

### CODE REVIEW

**Comment by kamilritz** [comment]
Yes, CPU load did not increase significantly.

File: EKF/ekf_helper.cpp [hunk 1]
```diff
@@ -1100,9 +1092,9 @@ bool Ekf::get_ekf_origin(uint64_t *origin_time, map_projection_reference_s *orig
 
 // return an array containing the output predictor angular, velocity and position tracking
```
[the remark this answers is not in the document]
Comment (kamilritz) [reply EKF/ekf_helper.cpp]: Done

**Comment by kamilritz** [comment]
@bresch Did some SITL testflights with MC and FW and checked the log files for mistakes. Seems good. Force pushed to rerun tests only. From my side this is ready to be merged.

**Comment by bresch** [comment]
I'll merge it and make a PR that introduce this and https://github.com/PX4/ecl/pull/770 for combined testing

**Pull request merged**